L1Trigger/L1THGCal/src/be_algorithms/HGCalMulticlusteringImpl.cc

```diff
@@ -95,6 +134,74 @@ void HGCalMulticlusteringImpl::clusterize( const edm::PtrVector<l1t::HGCalCluste
     }
     
 }
+void HGCalMulticlusteringImpl::clusterizeDBSCAN( const edm::PtrVector<l1t::HGCalCluster> & clustersPtrs, 
+                                           l1t::HGCalMulticlusterBxCollection & multiclusters)
+{
 
-
-
+  std::vector<l1t::HGCalMulticluster> multiclustersTmp;
+  l1t::HGCalMulticluster mcluTmp;
+  std::vector<bool> visited(clustersPtrs.size(),false);
+  std::vector<bool> merged (clustersPtrs.size(),false);
+  std::vector<std::vector<int>> neighborList;
+  int iclu = 0, imclu = 0;
+
+  for(edm::PtrVector<l1t::HGCalCluster>::iterator clu = clustersPtrs.begin(); clu != clustersPtrs.end(); ++clu, ++iclu){
+      std::vector<int> neighbors;      
+      
+      if(!visited.at(iclu)){
+	visited.at(iclu)=true;
```

Comment:
There are mixed tab and space indents from here down to the end of the function. Please use only spaces.